unstructured-mesh-part-ii.ipynb

```diff
@@ -142,13 +142,20 @@
    "metadata": {},
    "outputs": [
     {
-     "data": {
-      "text/plain": [
-       "'Downloading manifold.h5m: 100.0%'"
-      ]
-     },
-     "metadata": {},
-     "output_type": "display_data"
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "IOPub message rate exceeded.\n",
+      "The Jupyter server will temporarily stop sending output\n",
+      "to the client in order to avoid crashing it.\n",
+      "To change this limit, set the config variable\n",
+      "`--ServerApp.iopub_msg_rate_limit`.\n",
+      "\n",
+      "Current values:\n",
+      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
+      "ServerApp.rate_limit_window=3.0 (secs)\n",
```

Comment:
Any way we can avoid having this in the notebook?

Reply:
I think so, yeah. This was caused by the progress callback from `urlretrieve` overwhelming the notebook client. I've implemented a class with a `last_msg` to make sure that the `IPython.display.display` method is called only if the message is new. This seems to reduce the cell output updates enough to avoid this message.

Also, I tried relying on a variable in the `download` function as I assumed it would be part of the `progress_hook`'s enclosing scope, but while that was successful for me in a local interpreter it failed in the notebook so I went with the class/classmethod approach and relied on encapsulation.